capstone/src/main/java/capstone/Extras/Helper.java

```diff
@@ -139,4 +167,525 @@ public static void printLine(int size) {
+      ret[4] = nric;
+
+      String email = Helper.getInputWithValidation("Enter email", Helper::isValidEmail, false);
+      ret[5] = email;
+
+      while (true) {
+        System.out.print(String.format("%-50s: ", "Enter dateOfBirth in the format yyyy-mm-dd"));
+        String date = Helper.readLine();
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
+        try {
+          Date dateOfBirth = dateFormat.parse(date);
+          ret[6] = dateOfBirth;
+          break;
+        } catch (ParseException e) {
+          System.out.println(ANSI_RED + "Invalid input format. Please try again." + ANSI_RESET);
+        }
+      }
+
+      System.out.print(String.format("%-50s: ", "Enter address"));
+      String address = Helper.readLine();
+      ret[7] = address;
+
+      String phoneNumber =
+          Helper.getInputWithValidation("Enter phonenumber", Helper::isValidPhoneNumber, false);
+      ret[8] = phoneNumber;
+    }
+    return ret;
+  }
+  
+//TODO: TO replace this with printUserCredentials function in CustomerDisplayView and delete this fucntion
+  public static void printUserInfo(User user) {
+    System.out.println("Account Information " + "\u2B07");
+    Helper.printLine(80);
+    System.out.print(String.format("%-50s: %s%n", "userName", user.getUsername()));
+    System.out.print(String.format("%-50s: %s%n", "firstName", user.getFirstName()));
+    System.out.print(String.format("%-50s: %s%n", "lastName", user.getLastName()));
+  }
+  
+  /**
+   * Method that acquires inputs with validation.
+   * @param prompt guiding message to user for the input field
+   * @param validator function checks if the input is valid; return false if it's invalid
+   * @param isPassword true when the input field is a password.
+   */
+
+  public static String getInputWithValidation(
+      String prompt, Function<String, Boolean> validator, boolean isPassword) {
+    String input;
+    while (true) {
+      System.out.print(String.format("%-50s: ", prompt));
+      if (isPassword) input = Helper.getPasswordInput();
+      else input = Helper.readLine();
+
+      if (validator.apply(input)) {
+        return input;
+      } else {
+        System.out.println(ANSI_RED + "Invalid input format. Please try again." + ANSI_RESET);
+      }
+    }
+  }
+
+  /**
+   * Method that checks password pattern for validation.
+   * @param password the password input
+   */
+  public static boolean isValidPassword(String password) {
+    String passwordPattern = "^(?=.*[A-Z])(?=.*[a-z])(?=.*\\d)(?=.*[@#$%^&+=~`!]).{8,}$";
+    return password.matches(passwordPattern);
+  }
+
+  /**
+   * Method that checks nric pattern for validation.
+   * @param nric the nric input
+   */
+  public static boolean isValidNric(String nric) {
+    // NRIC format check: 9 characters with uppercase first and last characters
+    return nric.length() == 9
+        && Character.isUpperCase(nric.charAt(0))
+        && Character.isUpperCase(nric.charAt(8));
```

Comment:
better to use regex directly that checks if the string is `[GST]{1}[0-9]{7}[A-Z]{1}`. This is the basic check. The last character is supposed to be a checksum which is calculated as shown [here](https://userapps.support.sap.com/sap/support/knowledge/en/2572734)